src/routes/news/[...slug].svelte

```diff
@@ -0,0 +1,152 @@
+<script context="module">
+    import newsTargets from 'news-targets';
+
+    export async function preload({ params, query }) {
+        // the `slug` parameter is available because
+        // this file is called [slug].svelte
+        const [slug, title] = params.slug;
+        const res = await this.fetch(`news/${slug}.json`);
+        const data = await res.json();
+
+        // if title exists, try to find news with matching title. if not found, get the first one
+        // TODO: current support for multiple articles with same date is still hacky
+        let news =
+            (title && data.news.find((news) => news.title == title)) ||
+            data.news[0];
+
+        // https://github.com/rollup/rollup/issues/2463#issuecomment-455957865
```

Comment:
Similar to my previous comment, a short description of what this function is trying to do will be helpful